### DISCUSSION TITLE
Don't decompile all functions during initial auto analysis?

### DISCUSSION THREAD

**Discussion by krackers** [NationalSecurityAgency/ghidra#5443] [Q&A] [answered]
It seems like when initially loading a binary, Ghidra goes through and decompiles _all_ the functions. I'm only often interested in a single function, so this seems wasteful. Is there a way to have Ghidra only _disassemble_ but not _decompile_ all functions, and then later manually trigger decompilation for the functions that I care about?

Edit: The best option I can see so far is to set `Decompile Timeout` in tool options to `1`. This ensures the decompilation during automatic analysis times out, but succeeds when invoked manually. Setting to `0` does not work because based on https://github.com/NationalSecurityAgency/ghidra/blob/9ec89c37fb30d6228ed82d4cf1441b7a7aa0a3f1/Ghidra/Framework/Generic/src/main/java/ghidra/util/timer/GTimer.java#L42 value less than 0 causes it to never time out. Also only ints are allowed.

Edit: Setting to INT_MIN works better, because INT_MIN << 1 gives 0 (code multiplies by 1000 to convert to ms)

**Comment by astrelsky** [comment, marked as answer]
Simply turn off the decompiler parameter id, calling convention and switch analyzers. Basically all the analyzers that use the decompiler.

I'm assuming you meant decompiling functions during analysis and not when loading. Ghidra doesn't decompile or disassemble anything when loading and only processes the executable format.

**Reply by krackers** [reply]
Thank you! I also needed to turn off the objective-c decompiler message. A list of all such analyzers that make use of decompiler is those under https://github.com/NationalSecurityAgency/ghidra/tree/master/Ghidra/Features/Decompiler/src/main/java/ghidra/app/plugin/core/analysis
